windows/src/desktop/kmshell/main/Keyman.System.UpdateStateMachine.pas

```diff
@@ -996,17 +996,28 @@ function InstallingState.DoInstallPackage(PackageFileName: String): Boolean;
 var
   FPackage: IKeymanPackageFile2;
 begin
-  Result := True;
-  FPackage := kmcom.Packages.GetPackageFromFile(PackageFileName)
-    as IKeymanPackageFile2;
-  FPackage.Install2(True);
-  // Force overwrites existing package and leaves most settings for it intact
-  FPackage := nil;
-
-  kmcom.Refresh;
-  kmcom.Apply;
+  Result := False;
+  try
+    FPackage := kmcom.Packages.GetPackageFromFile(PackageFileName)
```

Comment:
Will `GetPackageFromFile` always return something valid, or do we have to add a check?

Reply:
`GetPackageFromFile` should not typically be crashing. It should be 'exception'al. So handling errors with exception handler is probably fine. But if there are errors happening regularly, we should be tracking it.